vowpalwabbit/nn.cc

```diff
@@ -262,7 +260,7 @@ void predict_or_learn_multi(nn& n, single_learner& base, example& ec)
       }
     }
 
-    n.all->loss = save_loss;
+    loss_function_swap_guard_converse_block.do_swap();
```

Comment:
do we need all the do_swaps technically? since they will go out of scope and trigger eventually, this might be more explicit though

Reply:
Perhaps not, but this code is pretty complicated and makes use of a `goto` so I didn't want to mess with the logic. This should be equivalent to the original code.